65.md

```diff
@@ -8,57 +8,114 @@ Relay List Metadata
+The event SHOULD include `relay-usage` tags, each with a relay URIs and a string indicating what the relay is used for. They should also include tags for backwards compatibility (see section below).
+
+`relay-usage` tags are in the following form:
+
+```json
+["relay-usage", "<relay-url>", "<usage-flags>"]
+```
+
+where `<usage-flags>` is a set of letters (in any order) consisting of any of the following:
+
+- `i` - INBOX
+- `o` - OUTBOX
+- `d` - DIRECT MESSAGES
+
+Clients should ignore any other letters found that they do not understand.
 
 The `.content` is not used.
 
+## Backwards Compatibility
+
+To maintain compatibility with clients which have not yet upgraded to this new format, tags of the old kind must also be included in these events.
+
+Such tags are of the following form
+```json
+["r", "<relay-url>", "<read-or-write>"]
+```
+
+`<read-or-write>` is optional and when missing means both inbox and outbox. Otherwise `<read-or-write>` MUST be either `read` (which means inbox) or `write` (which means outbox).
+
+## Example:
+
 ```json
 {
   "kind": 10002,
   "tags": [
+    ["relay-usage", "wss://alicerelay.example.com", "io"],
     ["r", "wss://alicerelay.example.com"],
+    ["relay-usage", "wss://brando-relay.com", "io"],
     ["r", "wss://brando-relay.com"],
+    ["relay-usage", "wss://expensive-relay.example2.com", "o"],
     ["r", "wss://expensive-relay.example2.com", "write"],
+    ["relay-usage", "wss://nostr-relay.example.com", "i"],
     ["r", "wss://nostr-relay.example.com", "read"],
+    ["relay-usage", "wss://secure.example.com", "d"],
   ],
   "content": "",
   ...other fields
 }
 ```
 
-This NIP doesn't fully replace relay lists that are designed to configure a client's usage of relays (such as `kind:3` style relay lists). Clients MAY use other relay lists in situations where a `kind:10002` relay list cannot be found.
+## The meaning of INBOX, OUTBOX and DIRECT MESSAGE
 
-## When to Use Read and Write Relays
+INBOX relays are for events that the relay list owner will be reading, specifically watching for events that tag them.
 
-When seeking events **from** a user, Clients SHOULD use the WRITE relays of the user's `kind:10002`.
+OUTBOX relays are for events that the relay list owner has authored and is publishing for public consumption.
 
-When seeking events **about** a user, where the user was tagged, Clients SHOULD use the READ relays of the user's `kind:10002`.
+DIRECT MESSAGE relays are like INBOX relays, except they are for reciept of DMS as they are considered more secure for this purpose by the relay list owner.
 
-When broadcasting an event, Clients SHOULD:
 
-- Broadcast the event to the WRITE relays of the author
-- Broadcast the event all READ relays of each tagged user
+### When to use which
```

Comment:
I miss NIP-65 guidance on which relays to pick when replying. Writing just to the guy I'm replying to's inbox relays (and my own outbox ones) and then everyone else repeating the same pattern would potentially spread the thread notes through a big set of relays. It would be very hard to fetch all thread notes.

I think we should leverage the fact that all replies must add the `e` tag with `root` marker (NIP-10).

What I'm doing is I download the `root` note to get the OP's pubkey > then I fetch it's `kind:10002` event > select the first 2 inbox relays (making the order relevant) and publish to them.

So I consider the first 2 OP's inbox relays as the "thread relays".

Reply:
IIRC there should be 'p' tags marking all the authors up the chain. For each of those 'p' tags you write to their inboxes.

Reply:
True! All good.